app/assets/src/components/common/DetailsSidebar/GeneDetailsMode/GeneDetailsMode.jsx

```diff
@@ -0,0 +1,297 @@
+
+    this.state = {
+      loading: true,
+      cardEntryFound: false,
+      collapseOntology: true,
+      ontology: null,
+    };
+  }
+
+  componentDidMount() {
+    this.getGeneInfo(this.props.geneName);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.geneName !== prevProps.geneName) {
+      this.setState({
+        loading: true,
+        cardEntryFound: false,
+        collapseOntology: true,
+      });
+      this.getGeneInfo(this.props.geneName);
+    }
+  }
+
+  async getGeneInfo(geneName) {
+    const ontology = await getCARDInfo(geneName);
+    const cardEntryFound = ontology.error !== "" ? false : true;
+
+    this.setState({
+      ontology,
+      loading: false,
+      cardEntryFound,
+    });
+  }
+
+  //*** Callback functions ***
+
+  expandOntology = () => {
+    this.setState({ collapseOntology: false });
+  };
+
+  //*** Functions depending on state ***
+
+  generateLinkTo(source) {
+    const { ontology } = this.state;
+    const { geneName } = this.props;
+    switch (source) {
+      case SOURCE_CARD: {
+        return URL_CARD_ARO + ontology.accession;
+      }
+      case SOURCE_OWL: {
+        return URL_CARD_OWL;
+      }
+      case SOURCE_PUBMED: {
+        return URL_PUBMED + "?term=" + geneName;
+      }
+      case SOURCE_GOOGLE_SCHOLAR: {
+        return URL_GOOGLE_SCHOLAR + geneName;
+      }
+      case SOURCE_NCBI_REF_GENE: {
+        return URL_NCBI_REF_GENE + geneName;
+      }
+      default: {
+        return "";
+      }
+    }
+  }
+
+  //*** Render methods ***
+
+  renderCARDLicense() {
+    return (
+      <div className={cs.cardLicense}>
+        This article uses material from the{" "}
+        <a
+          href={this.generateLinkTo(SOURCE_OWL)}
+          className={cs.cardLink}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          CARD Antibiotic Resistance Ontology
+        </a>, which is released under the{" "}
+        <a
+          href="https://creativecommons.org/licenses/by/4.0/"
+          className={cs.cardLink}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Creative Commons CC-BY license version 4.0
+        </a>{" "}
+        by McMaster University.
+      </div>
+    );
+  }
+
+  renderPubMedLicense() {
+    return (
+      <div className={cs.cardLicense}>
+        Publication names courtesy of the U.S. National Library of Medicine.
+      </div>
+    );
+  }
+
+  renderError() {
+    const { ontology } = this.state;
+    return <div className={cs.cardLicense}>{ontology.error}</div>;
+  }
+
+  renderHeader() {
+    const { loading, ontology } = this.state;
+    const { geneName } = this.props;
+    if (loading) {
+      return <div className={cs.loadingMsg}>Loading...</div>;
+    }
+    return (
+      <div className={cs.title}>
+        {ontology.label !== "" ? ontology.label : geneName}
+      </div>
+    );
+  }
+
+  renderOntology() {
+    const { ontology, collapseOntology } = this.state;
```

Comment:
this will be more DRY if you pull out the nested fields here like: `const { ontology: { synonyms, geneFamily, drugClass, publications, description }, collapseOntology } = this.state;`